drupal/web/modules/custom/covid/src/EventSubscriber/RequestEventSubscriber.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\covid\EventSubscriber;
+
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Drupal\user\Entity\User;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Listens to the request events.
+ */
+class RequestEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The messenger.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * RequestEventSubscriber constructor.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger.
+   */
+  public function __construct(RouteMatchInterface $route_match, AccountInterface $current_user, MessengerInterface $messenger) {
+    $this->routeMatch = $route_match;
+    $this->currentUser = $current_user;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * Checks one time password TFA is enabled for the current user.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
+   *   The request event.
+   */
+  public function checkOneTimePasswordEnabled(RequestEvent $event) {
+    // Only for authenticated users.
+    if ($this->currentUser->isAuthenticated()) {
+      $user = User::load($this->currentUser->id());
+
+      // Check if TFA is disabled.
+      if ($user->get('one_time_password')->isEmpty()) {
+
+        // Only if current route is not ignored.
+        if (!in_array($this->routeMatch->getRouteName(), [
+          'entity.user.edit_form',
```

Comment:
I think that reset password links are missing here, or is at that point user still anonymous?

Reply:
You are right. I checked login works fine, because the user/reset URL is anonymous, and then the user is redirected to user/edit, which is ignored already.